### ISSUE TITLE
Handle commented out code more gracefully

### ISSUE THREAD

**Issue by david-binda** [Automattic/VIP-Coding-Standards#181]
At least, we should not include `// End if().` `// End foreach().` etc.

Possibly, we could also just report on lines where the valid commented out code is the only thing on such line. Exempli gratia:

do not report as the line does not contain comment only:
```
} // End if().
```

report commented out line:
```
// if ( $a === $b ) {
```

**Comment by GaryJones** [comment]
I think the default value is 40% (if 40% of the string looks like a comment, it is recorded as a violation), but we could bump that to something like 70%.

**Comment by jrfnl** [comment]
The sniff which is used for this already has code in place to ignore things like `} // End if().`, but expects different spacing and capitalization `} //end if().`.

We _could_ submit a PR upstream to see if that check can be made little looser.

Other than that the sniff has seen quite some improvements in PHPCS 3.3.0 + 3.3.2 (June/September 2018), so the false positives previously seen, may no longer be reported.

**Comment by GaryJones** [comment]
I have to say that when reviewing customer code, I have seen fewer of these false positives (but there are still some violations), but I've also seen fewer `} End if.` comments anyway.

I see those types of "End" comments as an anti-pattern, so while not really a concern of VIPCS in terms of security, performance, or platform compatibility, not encouraging them by adjusting the commented-out-code sniff plays into the favour of our Code Quality initiative.
